util/patricia-trie-ethereum/src/rlp_node_codec.rs

```diff
@@ -70,65 +113,81 @@ impl NodeCodec<KeccakHasher> for RlpNodeCodec<KeccakHasher> {
 			_ => Err(DecoderError::Custom("Rlp is not valid."))
 		}
 	}
+
 	fn try_decode_hash(data: &[u8]) -> Option<<KeccakHasher as Hasher>::Out> {
-		let r = Rlp::new(data);
-		if r.is_data() && r.size() == KeccakHasher::LENGTH {
-			Some(r.as_val().expect("Hash is the correct size; qed"))
+
+		if data.len() == KeccakHasher::LENGTH {
```

Comment:
That is a change from upstream trie crate, the `try_decode_hash` is using directly the hash data, and we skip the header in `decode` function at 
`nodes[i] = Some(&value.as_raw()[1..]);`. (I will change this line, writing this comment make me realize there is  a cleanest way of doing this and without the implied malleability from skipping a byte).

The change is related to the fact that when trie uses this payload as an encoded children (`try_decode_hash` return none) trie code previously need to manage this byte.

So now this is done in `decode`. Thinking again that is not ideal and it would have been more direct to return either a hash or a new smaller slice from `try_decode` function.
Which is not perfect either (if the type of children could be encoded heading byte for instance).